src/library/dao/playlistdao.cpp

```diff
@@ -980,18 +981,19 @@ void PlaylistDAO::shuffleTracks(const int playlistId, const QList<int>& position
         trackPositionIds.insert(trackBPosition, trackAId);
         newPositions.swap(newPositions.indexOf(trackAPosition),
                           newPositions.indexOf(trackBPosition));
-        QString swapQuery = "UPDATE PlaylistTracks SET position=%1 "
-                "WHERE position=%2 AND playlist_id=%3";
-        query.exec(swapQuery.arg(QString::number(-1),
-                                 QString::number(trackAPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackAPosition),
-                                 QString::number(trackBPosition),
-                                 QString::number(playlistId)));
-        query.exec(swapQuery.arg(QString::number(trackBPosition),
-                                 QString::number(-1),
-                                 QString::number(playlistId)));
 
+        qSwap(tracksPositionRowIds[trackAPosition - 1].position,
+                  tracksPositionRowIds[trackBPosition - 1].position);
```

Comment:
@nimitbhardwaj : On one hand, you assume that position is "i + 1" on the other hand, you have a dedicated position field. Is the "i + 1" = position assumption still true? You original convinced me. 

If yes, you should revert your latest changes to https://github.com/mixxxdj/mixxx/pull/1205/commits/874d51180d981232a4fddeedbe03ebfeebf9e238

It was already working or have I missed something? 

The only issue was that the write back loop touches all playlist entry even though only a few are touched in a partly shuffling call. This issue still persists here.